### PR TITLE
Basic obstacle avoidance

### PR DESCRIPTION
**Pull request by VladLiteanu**
Very basic obstacle avoidance

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+    car.setAngle(90);
+    car.setSpeed(15);
+    delay(3000);
+    car.setAngle(0);
+    car.setSpeed(30);
+  }
+  if (frontIR.getDistance() > 30)
+  {
+    car.setSpeed(0);
+  }
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:36]: Are you sure this is the correct logic? Can you explain it to me?

Comment (VladLiteanu) [reply arduino/smartcar/smartcar.ino:36]: What it does is that it starts turning the car when the distance from an object is 170 and at the same time we slow it down. After 3 seconds we return the angle and the speed to the start values. 
If the car gets too close to an object (30), the car speed is set to 0 to not crash the car.

Comment (platisd) [reply arduino/smartcar/smartcar.ino:36]: What happens if the distance is 20?! Crash on it?

Comment (VladLiteanu) [reply arduino/smartcar/smartcar.ino]: True, we have updated the logic a bit.

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
+    car.setSpeed(15);
+    delay(3000);
+    car.setAngle(0);
+    car.setSpeed(30);
+	usDistance = front.getDistance();
+  }
+  if ((irDistance < 30) && (irDistance > 0))
+  {
+    car.setSpeed(0);
+	irDistance = frontIR.getDistance();
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:39]: Why is this needed?

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
+    car.setSpeed(30);
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+   int usDistance = front.getDistance();
+   int irDistance = frontIR.getDistance();
+  if((usDistance < 170) && (usDistance > 30) )
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:27]: What happens if `usDistance` is `0`, i.e. an error (no obstacle found)?

**Review by platisd — CHANGES_REQUESTED** [review]
Looks better now

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (VladLiteanu) [reply arduino/smartcar/smartcar.ino:39]: We wanted to use more than one sensor so this handles distances between 0 and 30 and stops the car.

File: arduino/smartcar/smartcar.ino (hunk 3 above)
Comment (VladLiteanu) [reply arduino/smartcar/smartcar.ino:27]: If there is no obstacle the car will just continue forward and if the distance is less than 30 then the IR sensor stops the car.

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (platisd) [reply arduino/smartcar/smartcar.ino:39]: I am referring to line 39: `irDistance = frontIR.getDistance();`

File: arduino/smartcar/smartcar.ino (hunk 3 above)
Comment (platisd) [reply arduino/smartcar/smartcar.ino:27]: I guess you are using the other sensor for smaller distances so that will work, OK

**Review by platisd — CHANGES_REQUESTED** [review]
The only issue is L39, but other than that it looks fine

File: arduino/smartcar/smartcar.ino [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+  int usDistance = front.getDistance();
+  int irDistance = frontIR.getDistance();
+  if((usDistance < 170) && (usDistance > 30))
+  {
+    car.setAngle(90);
+    car.setSpeed(15);
+    delay(3000);
+    car.setAngle(0);
+    car.setSpeed(30);
+    usDistance = front.getDistance();
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:34]: Now that i'm looking at this carefully, this isn't needed either :sweat_smile:

**Review by platisd — APPROVED** [review]

File: arduino/smartcar/smartcar.ino (hunk 4 above)
Comment (VladLiteanu) [reply arduino/smartcar/smartcar.ino]: We tested it just now and it works the same without it. We had some problems before and thought that maybe the distances didn't get updated.

Comment (platisd) [reply arduino/smartcar/smartcar.ino]: I mean, the new value is never used as it goes out of scope and a new variable is crated in the next `loop` iteration, so I would be extremely surprised if this had any effect on the behavior you're observing :sweat_smile:
